Cactooz/StarWarsWiki
# Add friends

New in this PR:

- Show/Hide User ID:
Users can now toggle the visibility of their ID with a button.
The button dynamically changes its label based on the current visibility status.

- Friend Requests Section:
Displays a section for incoming friend requests when there are pending requests.
Allows users to accept or decline friend requests individually.

- Add Friends:
Provides an input field for adding friends by entering their ID.
A custom message is displayed, if any, to provide additional context.

- Sent Requests Section:
Shows a section for sent friend requests, indicating users awaiting a response.
Users can cancel sent friend requests.

- Current Friends Section:
Displays a list of current friends with options to interact with each friend.
Users can navigate to a friend's profile or remove a friend from the list.

How to test:

Enter a user ID in the field and press Enter. If the ID is linked to a user a friend request will be sent and be visible as pending.
The user who got the request will see this as incoming friend request and choose to accept or decline. If declined the friend request is removed, aswell as the pending request. Otherwise you added each other as friends and can now check each others favorites.

TODO: 
- Styling
- Maybe add option to change nickname
